lib/src/Uncertainty/Algorithm/Sensitivity/HSICEstimatorImplementation.cxx

```diff
@@ -152,25 +152,37 @@ void HSICEstimatorImplementation::computePValuesPermutation() const
 {
   const SquareMatrix Wobs(computeWeightMatrix(outputSample_));
   PValuesPermutation_ = Point(inputDimension_);
+  Sample ShuffledSample;
+  Collection<Sample> shuffleCollection(permutationSize_);
+  Collection<SquareMatrix> weightMatrixCollection(permutationSize_);
+
+  for( UnsignedInteger b = 0; b < permutationSize_; ++b)
+  {
+    ShuffledSample = shuffledCopy(outputSample_);
```

Comment:
same here, the variable can be declared inside the loop
it should begin with with lowercase